app/celer-sim/Runner.cc

```diff
@@ -546,6 +550,44 @@ void Runner::build_step_collectors(RunnerInput const& inp)
     }
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Construct optical collector. This *MUST* be called *AFTER*
+ * \c this->build_core_params , as it depends on it.
+ */
+void Runner::build_optical_collector(RunnerInput const& inp,
+                                     ImportData const& imported)
+{
+    if (imported.optical.empty())
+    {
+        // No optical data loaded
+        return;
+    }
+
+    // TODO: Rethink if statements after implementing CelerOpticalPhysicsList
+    CELER_EXPECT(core_params_);
+    OpticalCollector::Input oc_inp;
+    auto const& optical_data = imported.optical.begin()->second;
+    if (static_cast<bool>(optical_data.properties))
+    {
+        oc_inp.properties = OpticalPropertyParams::from_import(imported);
+    }
+    if (static_cast<bool>(optical_data.properties))
+    {
+        oc_inp.cerenkov = std::make_shared<CerenkovParams>(oc_inp.properties);
+    }
+    if (static_cast<bool>(optical_data.scintillation))
```

Comment:
You don't need the static casts:
```suggestion
    if (optical_data.properties)
    {
        oc_inp.properties = OpticalPropertyParams::from_import(imported);
    }
    if (optical_data.properties)
    {
        oc_inp.cerenkov = std::make_shared<CerenkovParams>(oc_inp.properties);
    }
    if (optical_data.scintillation)
```
Also I don't know if the first properties/cerenkov should be combined. Maybe we need an optical model/process list...